MediaGalleryUi/Model/Directories/ExcludedDirectories.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUi\Model\Directories;
+
+/**
+ * Excluded directories validator
+ */
+class ExcludedDirectories
+{
+    /**
+     * @var array
+     */
+    private $excludedDirectories;
+
+    /**
+     * Constructor
+     *
+     * @param array $excludedDirectories
+     */
+    public function __construct(
+        array $excludedDirectories
+    ) {
+        $this->excludedDirectories = $excludedDirectories;
+    }
+
+    /**
+     * Validate if directory might be excluded
+     *
+     * @param string $path
+     * @return bool
+     */
+    public function validate(string $path): bool
```

Comment:
The result of this function is opposite to expected validation, in my opinion. Let's have more specific method name to avoid confusion
```suggestion
    public function isExcluded(string $path): bool
```